### DISCUSSION TITLE
callback to get to know that animations complete

### DISCUSSION THREAD

**Discussion by dannylin108** [corvudev/corvu#37] [Q&A] [answered]
is there a way to run my code after all corvu transitions complete?

for example `@ark-ui/solid` provides a callback `onExitComplete` for Dialog and Drawer primitives. How it's done in corvu?

And btw thanks a lot for this awesome project!

**Comment by GiyoMoon** [comment]
Hey! Currently you can use the `contentPresent` boolean which is available in the [context](https://corvu.dev/docs/primitives/dialog/#useContext) and [children callback](https://corvu.dev/docs/primitives/dialog/#RootChildrenProps). It only changes to false after the exit animations are complete.

But I agree that a dedicated function would be useful for this. I'm adding it to my list of todos 🙏 Thanks a lot for the idea and the kind words 😊

**Reply by GiyoMoon** [reply, marked as answer]
Hey @dannylin108! I just released new versions for the dialog, drawer and disclosure and they now accept at`onContentPresentChange` callback (or `onOverlayPresentChange` respectively) to get notified on presence changes :)
